pages/robot/tabs/components/index.php

```diff
@@ -284,6 +563,254 @@ function render_components(data) {
+                    description_id: description_id,
+                    record_id: record_id,
+                }));
+
+                // test ros service description
+                let _testDescriptionClient = new ROSLIB.Service({
+                    ros : window.ros['<?php echo $dbot_hostname ?>'],
+                    name : '/' + '<?php echo $robot_name?>' + '/' + component.test_service_name + '/description',
+                    serviceType : 'std_srvs/Trigger'
+                });
+                var reqDescription = new ROSLIB.ServiceRequest({});
+                _testDescriptionClient.callService(reqDescription, function(result) {
+                    console.log('Description service returned from \n'
+                    + _testDescriptionClient.name
+                    + '\n');
+                    // + result.message);
+
+                    // enable button
+                    $('#' + modal_btn_id).prop('disabled', false);
+                    
+                    try {
+                        // --- Method v1: send json and parse blocks
+                        const response_obj = JSON.parse(result.message);
+                        // console.log(response_obj);
+                        $('#' + description_id).html(json_to_html(response_obj));
+                        // --- Method v1 End
+                    } catch (error) {
+                        $('#' + description_id).html("Error parsing the response: " + result.message);
+                    }
+                });
+
+                // test ros service
+                let _testRunClient = new ROSLIB.Service({
+                    ros : window.ros['<?php echo $dbot_hostname ?>'],
+                    name : '/' + '<?php echo $robot_name?>' + '/' + component.test_service_name + '/run',
+                    serviceType : 'std_srvs/Trigger'
+                });
+                let request = new ROSLIB.ServiceRequest({});
+
+                // before test finish, do not show success/problem buttons
+                $('#' + btn_id_success).hide();
+
+                $('#' + btn_id_run).click(function() {
+                    console.log("Triggered " + id_str_name);
+
+                    // clear output
+                    $('#' + output_id).html("");
+                    // show progress
+                    $('#' + prog_id).show();
+                    // hide button
+                    $('#' + btn_id_run).hide();
+
+                    // console.log($dbot_hostname)
+                    _testRunClient.callService(request, function(result) {
+                        $('#' + prog_id).hide();
+                        $('#' + btn_id_run).show();
+                        // show result buttons
+                        $('#' + btn_id_success).show();
+                        console.log('Test run service returned from \n'
+                        + _testRunClient.name
+                        + '\n');
+                        // + result);
```

Comment:
remove

Reply:
This log I intend to keep, to indicate the test services finished from the ROS nodes.